index.js

```diff
@@ -10,8 +10,49 @@ var resolveRc = require('./lib/resolve-rc.js');
 var pkg = require('./package.json');
 var path = require('path');
 
+/**
+ * Error thrown by Babel formatted to conform to Webpack reporting.
+ */
+function BabelLoaderError(name, message, codeFrame, hideStack, error) {
+  Error.call(this);
+  Error.captureStackTrace(this, BabelLoaderError);
+
+  this.name = 'BabelLoaderError';
+  this.message = formatMessage(name, message, codeFrame);
+  this.hideStack = hideStack;
+  this.error = error;
+}
+
+BabelLoaderError.prototype = Object.create(Error.prototype);
+
+var STRIP_FILENAME_RE = /^[^:]+: /;
+
+var formatMessage = function(name, message, codeFrame) {
+  return (name ? name + ': ' : '') + message + '\n\n' + codeFrame + '\n';
+};
+
 var transpile = function(source, options) {
-  var result = babel.transform(source, options);
+  var result;
+  try {
+    result = babel.transform(source, options);
+  } catch (error) {
+    if (error.message && error.codeFrame) {
+      var message = error.message;
+      var name;
+      var hideStack;
```

Comment:
Actually we might make it `true` all the time. As stack is still shown if webpack is run with `--details` (or something like this). What do you think?

The only case this could be confusing is if someone debugs some Babel plugin with webpack.